FatPkg/EnhancedFatDxe/DiskCache.c

```diff
@@ -108,7 +331,7 @@ FatExchangeCachePage (
   UINTN       PageNo;
   UINTN       WriteCount;
   UINTN       RealSize;
-  UINT64      EntryPos;
+  UINTN       EntryPos;

```

Comment:
Why is this changed to UINTN from UINT64?  For 32-bit builds, does this limit to FAT volumes < 4GB?

Reply:
@mdkinney agreed, this does not need to be changed. This may have been needed in an earlier version of this series (otherwise MSVC starts adding IA32 compiler instrinsics for UINT64 math in places). I just tested with keeping that a UINT64 and the IA32 build passes, so I have removed the change.

Reply:
Ah, I did not test on MSVC, let me fix it up (not intrinsics, but type conversion)